src/images/minio.rs

```diff
@@ -0,0 +1,88 @@
+use crate::{core::WaitFor, Image, ImageArgs};
+use std::collections::HashMap;
+
+const NAME: &str = "minio/minio";
+const TAG: &str = "latest";
+
+const DIR: &str = "/data";
+const CONSOLE_ADDRESS: &str = ":9001";
+
+#[derive(Debug)]
+pub struct MinIO {
+    env_vars: HashMap<String, String>,
+    tag: String,
+}
+
+impl Default for MinIO {
+    fn default() -> Self {
+        Self::new_with_tag(TAG.to_owned())
+    }
+}
+
+impl MinIO {
+    pub fn new(tag: String, env_vars: HashMap<String, String>) -> Self {
+        Self { tag, env_vars }
+    }
+
+    pub fn new_with_tag(tag: String) -> Self {
+        let env_vars = HashMap::from([(
+            "MINIO_CONSOLE_ADDRESS".to_owned(),
+            CONSOLE_ADDRESS.to_owned(),
+        )]);
+        Self { tag, env_vars }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct MinIOServerArgs {
+    pub dir: String,
+    pub certs_dir: Option<String>,
+    pub json_log: bool,
+}
+
+impl Default for MinIOServerArgs {
+    fn default() -> Self {
+        Self {
+            dir: DIR.to_owned(),
+            certs_dir: None,
+            json_log: false,
+        }
+    }
+}
+
+impl ImageArgs for MinIOServerArgs {
+    fn into_iterator(self) -> Box<dyn Iterator<Item = String>> {
+        let mut args = vec!["server".to_owned(), self.dir.to_owned()];
+
+        if let Some(ref certs_dir) = self.certs_dir {
+            args.push("--certs-dir".to_owned());
+            args.push(certs_dir.to_owned())
+        }
+
+        if self.json_log {
+            args.push("--json".to_owned());
+        }
+
+        Box::new(args.into_iter())
+    }
+}
+
+impl Image for MinIO {
+    type Args = MinIOServerArgs;
+
+    fn name(&self) -> String {
+        NAME.to_owned()
+    }
+
+    fn tag(&self) -> String {
+        self.tag.to_owned()
+    }
+
+    fn ready_conditions(&self) -> Vec<WaitFor> {
+        vec![WaitFor::message_on_stdout("API:")]
```

Comment:
That seems odd. This really uniquely identifies a fully started container?

Reply:
<img width="421" alt="image" src="https://user-images.githubusercontent.com/26835520/155086373-be89b7b7-1236-410e-b6da-ea8822bf0974.png">

Reply:
Full output:
```sh
API: http://172.17.0.2:9000  http://127.0.0.1:9000 

Console: http://172.17.0.2:9001 http://127.0.0.1:9001 

Documentation: https://docs.min.io
```